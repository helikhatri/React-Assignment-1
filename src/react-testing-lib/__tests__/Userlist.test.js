import React from 'react';
import Login from "../../containers/Login";
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Users from '../../containers/Userlist';
import {render, fireEvent, cleanup} from '@testing-library/react';
import axios from 'axios';
import renderer from 'react-test-renderer';
//import Link from '../snapshot/Link.react';

jest.mock('axios');

Enzyme.configure({ adapter: new Adapter() });

describe('<DeleteRecord>', () => {
    it('Should delete particular user', () => {
        const wrapper = shallow(<Users />);
        expect(wrapper.find('input#btn')).toBeDefined();
    })

    it('should fetch users', () => {
        const users ='https://reqres.in/api/unknown';
        console.log(users);
        const resp = {data:users};
        axios.get.mockResolvedValue(resp);
        // expect(axiosMock.get).toHaveBeenCalledTimes(1);
        // expect(axiosMock.get).toHaveBeenCalledWith(users);
       // return Users.all().then(data => expect(data).toEqual(users));
    })

    it('renders correctly', () => {
        const tree = renderer
          .create(<Users>Userlist</Users>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
});


