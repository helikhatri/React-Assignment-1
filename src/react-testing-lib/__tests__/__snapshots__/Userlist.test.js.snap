// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteRecord> renders correctly 1`] = `
<div
  className="Home"
  style={
    Object {
      "float": "right",
      "width": "85%",
    }
  }
/>
`;
